infinite_tracing/test/infinite_tracing/worker_test.rb

```diff
@@ -25,6 +25,12 @@ def test_processes_simple_task
       assert_metrics_recorded "Supportability/InfiniteTracing/Worker"
     end
 
+    def test_worker_uses_agentthread
```

Comment:
I'd like for this test to guarantee that calling `Worker.new` actually triggers ` NewRelic::Agent::Threading::AgentThread.create`.

From what I can tell, the use of Mocha's `expects` here will yield `Thread.new` as desired in the event that `create` gets called, but there's no guarantee or assertion that it's being called.

To look at it another way, this test should fail when ran against an earlier version of the code base that did not cause `Worker.new` to create a new agent thread.

Reply:
That is how `.expects` works., it will fail if the method you are expecting is not called. Here is an example failure if I revert the worker change:
```
  1) Failure:
NewRelic::Agent::InfiniteTracing::ResponseHandlerTest#test_worker_uses_agentthread [/Users/tmcclure/projects/ruby-agent/infinite_tracing/test/infinite_tracing/worker_test.rb:29]:
not all expectations were satisfied
unsatisfied expectations:
- expected exactly once, not yet invoked: NewRelic::Agent::Threading::AgentThread.create(any_parameters)
```

Reply:
arg! Before I submitted my comment, I took the new test and added it to `worker_test.rb` in dev, ran the 8T tests, and everything passed. So I thought the test wasn't sufficient. My apologies - I must have not ran the tests correctly. As long as the test is dependent on the change being present (as it is), we're good. Thank you.